src/GeckofxHtmlToPDFComponent.cs

```diff
@@ -446,6 +462,158 @@ public void OnSecurityChange(nsIWebProgress aWebProgress, nsIRequest aRequest, u
 		}
 
 		#endregion
+
+		#region Ghostscript fiddling with output PDF
+
+		/// <summary>
+		/// The Mozilla code can produce HUGE PDF files especially if you have a number of color photograph files
+		/// displayed in the HTML.  So we use ghostscript to reduce the size of those files by redoing the image
+		/// data and compressing it.
+		/// </summary>
+		/// <returns>
+		/// path the the shrunk PDF file if successful, null if an error occurs or the file produced by ghostscript
+		/// is actually larger.  This file is in the temporary folder area.
+		/// </returns>
+		/// <remarks>
+		/// See http://issues.bloomlibrary.org/youtrack/issue/BL-3721.
+		/// </remarks>
+		private string CallGsToShrinkPdf()
+		{
+			RaiseStatusChanged(new PdfMakingStatus() { percentage = 0, statusLabel = "Shrinking PDF..." });
+			var newTempFile = GetTemporaryPdfFilename(); 
+			var info = new ProcessStartInfo
+			{
+				FileName = "gs",	// if installed on Linux, has simple name and will be in the path
+				Arguments = String.Format("-sDEVICE=pdfwrite -dBATCH -dNOPAUSE -dCompatibilityLevel=1.4 -dDownsampleColorImages=true -dColorImageResolution=600 -sOutputFile=\"{0}\" \"{1}\"",
+					newTempFile, _pathToTempPdf),
+				UseShellExecute = false,		// false for redirection
+				RedirectStandardOutput = true	// redirect for event handler reading asynchronously
+			};
+
+			try
+			{
+				Process gsProcess = new Process();
```

Comment:
I would recommend using the smarter process launcher used in Bloom (maybe libpalaso?) IIRC, that allows you to keep giving it more time so long as it's still outputting something?